editor.planx.uk/src/lib/dataMergedHotfix.ts

```diff
@@ -20,28 +20,44 @@ const getFlowData = async (id: string) => {
   return data.flows_by_pk;
 };
 
-export const dataMerged = async (
-  id: string,
-  ob: Record<string, any> = {},
-): Promise<Record<string, any>> => {
-  const { slug, data }: { slug: string; data: Record<string, any> } =
-    await getFlowData(id);
+// Flatten a flow's data to include main content & portals in a single JSON representation
+// XXX: getFlowData & dataMerged are currently repeated in api.planx.uk/helpers.ts
+//        in order to load frontend /preview routes for flows that are not published
+export const dataMerged = async (id: string, ob: Record<string, any> = {}) => {
+  // get the primary flow data
+  const { slug, data }: { slug: string; data: Record<string, any> } = await getFlowData(id);
+
+  // recursively get and flatten internal portals & external portals
   for (const [nodeId, node] of Object.entries(data)) {
-    if (nodeId === "_root" && Object.keys(ob).length > 0) {
+    const isExternalPortalRoot = nodeId === "_root" && Object.keys(ob).length > 0;
+    const isExternalPortal = node.type === TYPES.ExternalPortal;
+    const isMerged = ob[node.data?.flowId];
```

Comment:
nit: is `isNotMerged` a more clear name here since `node.data?.flowId` indicates it's still an external portal reference, _not_ flattened representation of individual nodes?

Reply:
So here `ob` is the returned item (the flattened flow). If `ob[node.data?.flowId] == true` then the flow has already been flattened and `ob[example-uuid-of-portal]` exists. Does that make sense?

Reply:
TBH I think some of the language here is a little fuzzy around merging / flattening